src/components/Menu/MenuItem/MenuItem.tsx

```diff
@@ -117,9 +117,12 @@ const MenuItem: VibeComponent<MenuItemProps> & {
     const isSubMenuOpen = !!children && isActive && hasOpenSubMenu;
     const hasChildren = !!children;
     const shouldShowSubMenu = hasChildren && isParentMenuVisible && isSubMenuOpen;
-    const submenuChild = children && React.Children.only(children);
+    const submenuChild =
+      children &&
+      (React.Children.only(children) as ReactElement & {
+        type: Record<string, unknown>;
+      });
```

Comment:
Let's revert this change and instead change type of `children` prop:
```diff
-   children?: ReactElement | ReactElement[];
+   children?: MenuChild | MenuChild[];
```

Reply:
```suggestion
    const submenuChild: MenuChild = children && React.Children.only(children);
```
Cause still React.ChildrenOnly will return ReactElement